src/main/kotlin/org/rust/lang/core/psi/RsCodeFragment.kt

```diff
@@ -111,3 +127,29 @@ class RsTypeReferenceCodeFragment(project: Project, text: CharSequence, context:
       RsNamedElement {
     val typeReference: RsTypeReference? get() = PsiTreeUtil.getChildOfType(this, RsTypeReference::class.java)
 }
+
+class RsPathCodeFragment(
+    fileViewProvider: FileViewProvider,
+    context: RsElement,
+    mode: RustParserUtil.PathParsingMode,
+    val ns: Set<Namespace>
+) : RsCodeFragment(fileViewProvider, mode.elementType(), context), RsInferenceContextOwner {
+    constructor(
+        project: Project,
+        text: CharSequence,
+        eventSystemEnabled: Boolean,
+        context: RsElement,
+        mode: RustParserUtil.PathParsingMode,
+        ns: Set<Namespace>
+    ) : this(createFileViewProvider(project, text, eventSystemEnabled), context, mode, ns)
+
+    val path: RsPath? get() = PsiTreeUtil.getChildOfType(this, RsPath::class.java)
+
+    companion object {
+        @JvmStatic
+        private fun RustParserUtil.PathParsingMode.elementType() = when (this) {
+            RustParserUtil.PathParsingMode.NO_COLONS -> RsCodeFragmentElementType.PATH_WITHOUT_COLONS
+            else -> RsCodeFragmentElementType.PATH_WITH_COLONS
```

Comment:
Maybe we want to add path code fragment type for `RustParserUtil.PathParsingMode.NO_TYPES` too?
Because now it's possible to pass `NO_TYPES` as `mode` and for this case we will use unexpected parsing rool

Reply:
@vlad20012 I still can pass `NO_TYPES` into `RsPathCodeFragment` constructor

Reply:
fixed